src/psyclone/tests/psyir/transformations/arraynotation2arrayrange_trans_test.py

```diff
@@ -0,0 +1,199 @@
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Author R. W. Ford
+
+'''Module containing tests for the ArrayNotation2ArrayRangeLoopTrans
+transformation.'''
+
+import pytest
+
+from psyclone.psyir.nodes import Reference
+from psyclone.psyGen import Transformation
+from psyclone.psyir.symbols import DataSymbol, REAL_TYPE
+from psyclone.psyir.transformations import ArrayNotation2ArrayRangeTrans, \
+    TransformationError
+
+
+CODE = (
+    "program test\n"
+    "  real, dimension(10) :: a\n"
+    "  real :: b\n\n"
+    "  a = b\n"
+    "end program test\n")
+
+
+def apply_trans(fortran_reader, fortran_writer, code):
+    '''Utility function that applies the transformation to the code.
+
+    :param fortran_reader: a fortran reader object
+    :type fortran_reader: \
+        :py:class:`psyclone.psyir.frontend.fortran.FortranReader`
+    :param fortran_writer: a fortran writer object
+    :type fortran_writer: \
+        :py:class:`psyclone.psyir.backend.fortran.FortranWriter`
+    :param str code: the Fortran input code.
+
+    returns: the transformed Fortran code.
+    :rtype str
+
+    '''
+    trans = ArrayNotation2ArrayRangeTrans()
+    psyir = fortran_reader.psyir_from_source(code)
+    for reference in psyir.walk(Reference):
+        try:
+            trans.apply(reference)
+        except TransformationError:
+            pass
+    return fortran_writer(psyir)
+
+
+def test_get_array_bound(fortran_reader):
+    ''' Test that the get_array_bound utility works as expected '''
+    # known bounds
+    psyir = fortran_reader.psyir_from_source(CODE)
+    node = psyir.walk(Reference)[0]
+    symbol = node.symbol
+    lower_bound, upper_bound, step = \
+        ArrayNotation2ArrayRangeTrans._get_array_bound(symbol, 0)
+    from psyclone.psyir.nodes import Literal, BinaryOperation
+    assert isinstance(lower_bound, Literal)
+    assert lower_bound.value == "1"
+    assert isinstance(upper_bound, Literal)
+    assert upper_bound.value == "10"
+    assert isinstance(step, Literal)
+    assert step.value == "1"
+    # unknown bounds
+    psyir = fortran_reader.psyir_from_source(
+        CODE.replace("dimension(10)", "dimension(:)"))
+    node = psyir.walk(Reference)[0]
+    symbol = node.symbol
+    lower_bound, upper_bound, step = \
+        ArrayNotation2ArrayRangeTrans._get_array_bound(symbol, 0)
+    assert isinstance(lower_bound, BinaryOperation)
+    assert lower_bound.operator == BinaryOperation.Operator.LBOUND
+    assert isinstance(upper_bound, BinaryOperation)
+    assert upper_bound.operator == BinaryOperation.Operator.UBOUND
+    assert isinstance(step, Literal)
+    assert step.value == "1"
```

Comment:
For completeness, please add a test for an idx other than zero. As it's an internal method I won't ask for error checking on the value of `idx`.

Reply:
Added test for idx==1.